iron-location.html

```diff
@@ -181,7 +181,11 @@
             this.path).replace(/\#/g, '%23').replace(/\?/g, '%3F');
         var partiallyEncodedQuery = '';
         if (this.query) {
-          partiallyEncodedQuery = '?' + this.query.replace(/\#/g, '%23');
+          // decoding here needed to prevent encoding the % sign. decodeURIComponent
+          // because it will not decode the hash otherwise. encodeURI and not
+          // component because it will encode ? to %3F otherwise.
```

Comment:
Why wouldn't you encode `?` in the query as well? `?` is used to mark the query params in an URL, so it wouldn't make sense to have more than one.

If you do use `encodeURIComponent`, it will also encode `#` so you won't need that replace method.